apps/ewallet/lib/ewallet/gates/mint_gate.ex

```diff
@@ -31,8 +31,12 @@ defmodule EWallet.MintGate do
     mint_token(token, attrs)
   end
 
-  def mint_token(%{blockchain_address: address}, _) when not is_nil(address) do
-    {:error, :invalid_parameter, "A blockchain-enabled token cannot be minted."}
+  def mint_token(%{blockchain_address: address} = token, _) when not is_nil(address) do
+    with true <- Token.blockchain_confirmed?(token) || {:error, :token_is_not_confirmed} do
```

Comment:
I think in this specific code, case is easier to understand.

```elixir
case Token.blockchain_confirmed?(token) do
  true ->
    {:error, :invalid_parameter, "A blockchain-enabled token cannot be minted."}

  false ->
    {:error, :token_is_not_confirmed}
end
```

Which seems to me, we can even simply return the "blockchain-enabled token cannot be minted" error for both cases?

Reply:
I'll go with the case, this was just to prepare for next PR where we will be able to mint blockchain tokens. Just so wee don't forget to check for token status :)

Reply:
Oh ok I didn't realize it's part of next PR. Makes sense to go with with clauses in that case.